lib/Driver/ToolChains.cpp

```diff
@@ -1468,6 +1468,56 @@ void ToolChain::addLinkRuntimeLib(const ArgList &Args, ArgStringList &Arguments,
   Arguments.push_back(Args.MakeArgString(P));
 }
 
+void ToolChain::addPluginArguments(const ArgList &Args,
+                                   ArgStringList &Arguments) const {
+  auto swiftProgramPath = getDriver().getSwiftProgramPath();
+#if defined(_WIN32)
+  SmallString<261> pluginPathBase = StringRef(swiftProgramPath);
+  llvm::sys::path::remove_filename(pluginPathBase); // Remove `swift`
+#elif defined(__APPLE__) || defined(__unix__)
+  SmallString<64> pluginPathBase;
+  CompilerInvocation::computeRuntimeResourcePathFromExecutablePath(
+      swiftProgramPath, /*shared=*/true, pluginPathBase);
+#else
```

Comment:
I checked  `swift-driver` implementation
https://github.com/apple/swift-driver/blob/main/Sources/SwiftDriver/Jobs/FrontendJobHelpers.swift#L793-L802
Since it's all based on the executable directory. Let's align with it.

Reply:
And yes, adding `pluginPathRoot.localPluginPath` in Windows is not correct 😅 
https://github.com/apple/swift-driver/blob/6d1e5d2bd8e67432176cd843704bd2277616376f/Sources/SwiftDriver/Jobs/FrontendJobHelpers.swift#L810